From 041c0aba21ea3f85976cd1c46edff49bf7218053 Mon Sep 17 00:00:00 2001
From: Jeff Peeler <jpeeler@gmail.com>
Date: Wed, 14 Mar 2012 11:43:22 -0400
Subject: [PATCH] build: Don't autogenerate man pages, use copies in source
 directory temporarily.
*Modified slightly to remove dbus-bridge, which isn't present here.

This adds a new cmake macro to handle installing man pages if one exists
in the source directory and adds a man page for every binary. The change is
temporary so that man pages are installed while autogeneration is not working.

Once autogeneration of man pages is working again, all the CMakeLists
changes can be changed back with the exception of the broker agent which will
never be autogenerated (the new install_manpage macro could stay too).

Signed-off-by: Jeff Peeler <jpeeler@redhat.com>
---
 .gitignore                                      |    1 +
 src/.gitignore                                  |    1 +
 src/broker/CMakeLists.txt                       |   13 ++-
 src/cmake/modules/MatahariMacros.cmake          |   14 +++
 src/host/CMakeLists.txt                         |    3 +-
 src/host/matahari-qmf-hostd.8                   |   79 ++++++++++++++
 src/network/CMakeLists.txt                      |    3 +-
 src/network/matahari-qmf-networkd.8             |   79 ++++++++++++++
 src/rpc/CMakeLists.txt                          |    3 +-
 src/rpc/matahari-qmf-rpcd.8                     |   79 ++++++++++++++
 src/service/CMakeLists.txt                      |    6 +-
 src/service/matahari-qmf-service-cli.8          |  133 +++++++++++++++++++++++
 src/service/matahari-qmf-serviced.8             |   79 ++++++++++++++
 src/sysconfig/CMakeLists.txt                    |    6 +-
 src/sysconfig/matahari-qmf-sysconfig-consoled.8 |   79 ++++++++++++++
 src/sysconfig/matahari-qmf-sysconfigd.8         |   79 ++++++++++++++
 18 files changed, 725 insertions(+), 14 deletions(-)
 create mode 100644 src/host/matahari-qmf-hostd.8
 create mode 100644 src/network/matahari-qmf-networkd.8
 create mode 100644 src/rpc/matahari-qmf-rpcd.8
 create mode 100644 src/service/matahari-qmf-service-cli.8
 create mode 100644 src/service/matahari-qmf-serviced.8
 create mode 100644 src/sysconfig/matahari-qmf-sysconfig-consoled.8
 create mode 100644 src/sysconfig/matahari-qmf-sysconfigd.8

diff --git a/.gitignore b/.gitignore
index 33b2db7..14f8e32 100644
--- a/.gitignore
+++ b/.gitignore
@@ -17,5 +17,6 @@ mock
 *.orig
 *.rej
 *.rpm
+*.swp
 tags
 TAGS
diff --git a/src/.gitignore b/src/.gitignore
index 29b9456..0d284b3 100644
--- a/src/.gitignore
+++ b/src/.gitignore
@@ -4,6 +4,7 @@ config.h
 include/matahari.h
 qmf
 matahari-*
+!matahari-*.8
 *-dbus-glue.h
 *-dbus-properties.h
 org.matahariproject.xml
diff --git a/src/broker/CMakeLists.txt b/src/broker/CMakeLists.txt
index f5959c6..c688316 100644
--- a/src/broker/CMakeLists.txt
+++ b/src/broker/CMakeLists.txt
@@ -12,12 +12,13 @@ if(NOT WIN32)
     install(TARGETS ${BROKER_DAEMON} DESTINATION sbin)
 
 
-    SET(MAN_PAGE ${BROKER_DAEMON}.${AGENT_MAN_SECTION})
-    install(CODE "execute_process(COMMAND gzip
-        INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${MAN_PAGE}
-        OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/${MAN_PAGE}.gz)")
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${MAN_PAGE}.gz
-        DESTINATION share/man/man${AGENT_MAN_SECTION})
+    #SET(MAN_PAGE ${BROKER_DAEMON}.${AGENT_MAN_SECTION})
+    #install(CODE "execute_process(COMMAND gzip
+    #    INPUT_FILE ${CMAKE_CURRENT_SOURCE_DIR}/${MAN_PAGE}
+    #    OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/${MAN_PAGE}.gz)")
+    #install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${MAN_PAGE}.gz
+    #    DESTINATION share/man/man${AGENT_MAN_SECTION})
+    install_manpage(${BROKER_DAEMON} ${AGENT_MAN_SECTION})
 
     configure_file(${CMAKE_SOURCE_DIR}/sys/matahari-broker.init.in
                    ${CMAKE_CURRENT_BINARY_DIR}/matahari-broker.init @ONLY)
diff --git a/src/cmake/modules/MatahariMacros.cmake b/src/cmake/modules/MatahariMacros.cmake
index 615b010..de1f1ae 100644
--- a/src/cmake/modules/MatahariMacros.cmake
+++ b/src/cmake/modules/MatahariMacros.cmake
@@ -119,6 +119,20 @@ macro(create_manpage BINARY SECTION DESC)
     endif(NOT WIN32)
 endmacro(create_manpage)
 
+macro(install_manpage BINARY SECTION)
+    SET(MAN_PAGE_IN ${CMAKE_CURRENT_SOURCE_DIR}/${BINARY}.${SECTION})
+    SET(MAN_PAGE_OUT ${CMAKE_CURRENT_BINARY_DIR}/${BINARY}.${SECTION}.gz)
+
+    if(NOT EXISTS ${MAN_PAGE_OUT})
+        execute_process(COMMAND gzip INPUT_FILE ${MAN_PAGE_IN} OUTPUT_FILE ${MAN_PAGE_OUT})
+        Message("Prepared ${BINARY} man page")
+    endif(NOT EXISTS ${MAN_PAGE_OUT})
+
+    if(NOT WIN32)
+        install(FILES ${MAN_PAGE_OUT} DESTINATION share/man/man${SECTION})
+    endif(NOT WIN32)
+endmacro(install_manpage)
+
 macro(create_service_scripts BASE)
     if(NOT WIN32)
         configure_file(${CMAKE_SOURCE_DIR}/sys/matahari.init.in ${CMAKE_CURRENT_BINARY_DIR}/matahari-${BASE})
diff --git a/src/host/CMakeLists.txt b/src/host/CMakeLists.txt
index dd92d3f..2514e99 100644
--- a/src/host/CMakeLists.txt
+++ b/src/host/CMakeLists.txt
@@ -12,7 +12,8 @@ if(WITH-QMF)
     add_executable(${QMF_AGENT} ${BASE}-qmf.cpp ${SCHEMA_SOURCES})
     target_link_libraries(${QMF_AGENT} ${BASE_LIB} mcommon_qmf)
 
-    create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    #create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    install_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION})
     create_service_scripts(${BASE})
 
     install(TARGETS ${QMF_AGENT} DESTINATION sbin)
diff --git a/src/host/matahari-qmf-hostd.8 b/src/host/matahari-qmf-hostd.8
new file mode 100644
index 0000000..f2b93e0
--- /dev/null
+++ b/src/host/matahari-qmf-hostd.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-HOSTD "8" "March 2012" "matahari-hostd <options>" "System Administration Utilities"
+.SH NAME
+matahari-hostd \- Part of the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-hostd <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/network/CMakeLists.txt b/src/network/CMakeLists.txt
index 8f4313f..6b1809b 100644
--- a/src/network/CMakeLists.txt
+++ b/src/network/CMakeLists.txt
@@ -12,7 +12,8 @@ if(WITH-QMF)
     add_executable(${QMF_AGENT} ${BASE}-qmf.cpp ${SCHEMA_SOURCES})
     target_link_libraries(${QMF_AGENT} ${BASE_LIB} mcommon_qmf)
 
-    create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    #create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    install_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION})
     create_service_scripts(${BASE})
 
     install(TARGETS ${QMF_AGENT} DESTINATION sbin)
diff --git a/src/network/matahari-qmf-networkd.8 b/src/network/matahari-qmf-networkd.8
new file mode 100644
index 0000000..6bd5206
--- /dev/null
+++ b/src/network/matahari-qmf-networkd.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-NETWORKD "8" "March 2012" "matahari-Networkd <options>" "System Administration Utilities"
+.SH NAME
+matahari-Networkd \- Part of the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-Networkd <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/rpc/CMakeLists.txt b/src/rpc/CMakeLists.txt
index f5595fb..40c37dd 100644
--- a/src/rpc/CMakeLists.txt
+++ b/src/rpc/CMakeLists.txt
@@ -12,7 +12,8 @@ if(NOT WIN32)
     add_executable(${QMF_AGENT} ${BASE}-qmf.cpp ${SCHEMA_SOURCES})
     target_link_libraries(${QMF_AGENT} ${BASE_LIB} mcommon_qmf)
 
-    create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    #create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    install_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION})
     create_service_scripts(${BASE})
 
     install(TARGETS ${QMF_AGENT} DESTINATION sbin)
diff --git a/src/rpc/matahari-qmf-rpcd.8 b/src/rpc/matahari-qmf-rpcd.8
new file mode 100644
index 0000000..6b320b6
--- /dev/null
+++ b/src/rpc/matahari-qmf-rpcd.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-RPCD "8" "March 2012" "matahari-rpcd <options>" "System Administration Utilities"
+.SH NAME
+matahari-rpcd \- Part of the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-rpcd <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/service/CMakeLists.txt b/src/service/CMakeLists.txt
index 586ed68..ae1bc31 100644
--- a/src/service/CMakeLists.txt
+++ b/src/service/CMakeLists.txt
@@ -16,8 +16,10 @@ if(WITH-QMF)
     add_executable(${QMF_CONSOLE} ${BASE}-qmf-console.cpp)
     target_link_libraries(${QMF_CONSOLE} mcommon_qmf)
 
-    create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
-    create_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION} ${CONSOLE_MAN_DESC})
+    #create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    #create_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION} ${CONSOLE_MAN_DESC})
+    install_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION})
+    install_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION})
     create_service_scripts(${BASE})
 
     install(TARGETS ${QMF_AGENT} DESTINATION sbin)
diff --git a/src/service/matahari-qmf-service-cli.8 b/src/service/matahari-qmf-service-cli.8
new file mode 100644
index 0000000..93443f9
--- /dev/null
+++ b/src/service/matahari-qmf-service-cli.8
@@ -0,0 +1,133 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-QMF-SERVICE-CLID "8" "March 2012" "matahari-qmf-service-clid <options>" "System Administration Utilities"
+.SH NAME
+matahari-qmf-service-clid \- Console for the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-qmf\-service\-clid <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-A\fR | \fB\-\-agent\fR
+(Resources API only)
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-H\fR | \fB\-\-host\-dns\fR
+Host DNS name
+.TP
+\fB\-J\fR | \fB\-\-provider\fR
+(Resources API only)
+.TP
+\fB\-N\fR | \fB\-\-name\fR
+Name of a resource
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-S\fR | \fB\-\-standard\fR
+lsb|ocf|windows (Resources API only)
+.TP
+\fB\-T\fR | \fB\-\-api\-type\fR
+Resources|Services
+.TP
+\fB\-U\fR | \fB\-\-host\-uuid\fR
+Host UUID
+.TP
+\fB\-a\fR | \fB\-\-action\fR
+Action to perform
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-i\fR | \fB\-\-interval\fR
+(Resources API only) Time, in milliseconds, in between recurring executions of this action.
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-o\fR | \fB\-\-option\fR
+Option to pass to the resource script (Resources API only)
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-A\fR | \fB\-\-agent\fR
+(Resources API only)
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-H\fR | \fB\-\-host\-dns\fR
+Host DNS name
+.TP
+\fB\-J\fR | \fB\-\-provider\fR
+(Resources API only)
+.TP
+\fB\-N\fR | \fB\-\-name\fR
+Name of a resource
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-S\fR | \fB\-\-standard\fR
+lsb|ocf|windows (Resources API only)
+.TP
+\fB\-T\fR | \fB\-\-api\-type\fR
+Resources|Services
+.TP
+\fB\-U\fR | \fB\-\-host\-uuid\fR
+Host UUID
+.TP
+\fB\-a\fR | \fB\-\-action\fR
+Action to perform
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-i\fR | \fB\-\-interval\fR
+(Resources API only) Time, in milliseconds, in between recurring executions of this action.
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-o\fR | \fB\-\-option\fR
+Option to pass to the resource script (Resources API only)
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/service/matahari-qmf-serviced.8 b/src/service/matahari-qmf-serviced.8
new file mode 100644
index 0000000..66cab07
--- /dev/null
+++ b/src/service/matahari-qmf-serviced.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-SERVICED "8" "March 2012" "matahari-serviced <options>" "System Administration Utilities"
+.SH NAME
+matahari-serviced \- Part of the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-serviced <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/sysconfig/CMakeLists.txt b/src/sysconfig/CMakeLists.txt
index 9a491d1..945fa80 100644
--- a/src/sysconfig/CMakeLists.txt
+++ b/src/sysconfig/CMakeLists.txt
@@ -16,8 +16,10 @@ if(WITH-QMF)
     target_link_libraries(${QMF_AGENT} ${BASE_LIB} mcommon_qmf)
     target_link_libraries(${QMF_CONSOLE} mcommon_qmf)
 
-    create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
-    create_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION} ${CONSOLE_MAN_DESC})
+    #create_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION} ${AGENT_MAN_DESC})
+    #create_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION} ${CONSOLE_MAN_DESC})
+    install_manpage(${QMF_AGENT} ${AGENT_MAN_SECTION})
+    install_manpage(${QMF_CONSOLE} ${CONSOLE_MAN_SECTION})
     create_service_scripts(${BASE})
 
     set(BASE_SUB "-console")
diff --git a/src/sysconfig/matahari-qmf-sysconfig-consoled.8 b/src/sysconfig/matahari-qmf-sysconfig-consoled.8
new file mode 100644
index 0000000..aedcb4b
--- /dev/null
+++ b/src/sysconfig/matahari-qmf-sysconfig-consoled.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-SYSCONFIG-CONSOLED "8" "March 2012" "matahari-sysconfig-consoled <options>" "System Administration Utilities"
+.SH NAME
+matahari-sysconfig-consoled \- Console for the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-sysconfig\-consoled <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
diff --git a/src/sysconfig/matahari-qmf-sysconfigd.8 b/src/sysconfig/matahari-qmf-sysconfigd.8
new file mode 100644
index 0000000..f8b36e1
--- /dev/null
+++ b/src/sysconfig/matahari-qmf-sysconfigd.8
@@ -0,0 +1,79 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.6.
+.TH MATAHARI-SYSCONFIGD "8" "March 2012" "matahari-Sysconfigd <options>" "System Administration Utilities"
+.SH NAME
+matahari-Sysconfigd \- Part of the Matahari suite of remote agents
+.SH DESCRIPTION
+matahari\-Sysconfigd <options>
+.SH OPTIONS
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
+.TP
+\fB\-h\fR | \fB\-\-help\fR
+print this help message.
+.TP
+\fB\-D\fR | \fB\-\-dns\-srv\fR
+interpret the value of \fB\-\-broker\fR as a domain name for DNS SRV lookups
+.TP
+\fB\-P\fR | \fB\-\-password\fR
+password to use for authentication to the broker
+.TP
+\fB\-b\fR | \fB\-\-broker\fR
+specify broker host name
+.TP
+\fB\-d\fR | \fB\-\-daemon\fR
+run as a daemon
+.TP
+\fB\-n\fR | \fB\-\-ssl\-cert\-name\fR
+name of the certificate to use
+.TP
+\fB\-p\fR | \fB\-\-port\fR
+specify broker port
+.TP
+\fB\-r\fR | \fB\-\-reconnect\fR
+attempt to reconnect if the broker connection is lost
+.TP
+\fB\-s\fR | \fB\-\-service\fR
+service name to use for authentication to the broker
+.TP
+\fB\-t\fR | \fB\-\-use\-tls\fR
+Use TLS/SSL encryption
+.TP
+\fB\-u\fR | \fB\-\-username\fR
+username to use for authentication to the broker
+.TP
+\fB\-v\fR | \fB\-\-verbose\fR
+Increase the log level
-- 
1.7.7.6

